### PR TITLE
Add `localStorage` `sessionStorage` to cdpfn

### PR DESCRIPTION
**Pull request by atsushi-ishibashi**
# Overview
access to localStorage and sessionStorage.
This enables you
- check storage state
- get data in localStorage or sessionStorage which is difficult to produce via http, for example authentication sdk output

### Working example
```
// add debug log
for _, v := range resp {
	log.Println("localStorage: len: %d, item: %v", len(v), v)
	if len(v) != 2 {
		continue
	}
	m[v[0]] = v[1]
}
```
```
//workbook.yml
desc: Browser Storage
runners:
  cc: chrome://new
steps:
  google:
    cc:
      actions:
        - navigate: https://www.google.com/
        - localStorage:
            origin: https://www.google.com
        - sessionStorage:
            origin: https://www.google.com
```

```
$ runn run workbook.yml --debug

Run 'cc' on 'Browser Storage'.steps.google
>>>>>START CDP>>>>>
-----START CDP ACTION-----
name: navigate
args:
url: "https://www.google.com/"
-----END CDP ACTION-----
-----START CDP ACTION-----
name: localStorage
args:
origin: "https://www.google.com"
-----END CDP ACTION-----
2022/11/25 13:33:04 localStorage: len: %d, item: %v 2 [sb_wiz.zpc.gws-wiz. {"0":[["関東大学ラグビー対抗戦早慶戦",0,[143,362],{"zf":33,"zl":8,"zp":{"gs_ss":"1"}}],["ペイペイカードゴールド",0,[143,362],{"zf":33,"zl":8,"zp":{"gs_ss":"1"}}],["サッカー 浅野 弟",0,[143,362],{"zf":33,"zl":8,"zp":{"gs_ss":"1"}}],["ポケモン sv かがやきパワーレシピ",0,[143,362],{"zf":33,"zl":8,"zp":{"gs_ss":"1"}}],["ポケモン sv 夢特性入手方法",0,[143,362],{"zf":33,"zl":8,"zp":{"gs_ss":"1"}}],["熊本地震速報震源地",0,[143,362],{"zf":33,"zl":8,"zp":{"gs_ss":"1"}}],["スバル レヴォーグ",0,[143,362],{"zf":33,"zl":8,"zp":{"gs_ss":"1"}}],["mlb news",0,[143,362],{"zf":33,"zl":8,"zp":{"gs_ss":"1"}}],["帝京大学ゼミ男女差別教授",0,[143,362],{"zf":33,"zl":8,"zp":{"gs_ss":"1"}}],["名鉄名古屋本線踏切事故",0,[143,362],{"zf":33,"zl":8,"zp":{"gs_ss":"1"}}]],"1":{"ag":{"a":{"8":["話題の検索キーワード"]}},"k":1,"q":"b-8V15fiDAldvMgM8dUmVen0kpY"}}]
-----START CDP RESPONSE-----
name: localStorage
response:
items: {"sb_wiz.zpc.gws-wiz.":"{\"0\":[[\"関東大学ラグビー対抗戦早慶戦\",0,[143,362],{\"zf\":33,\"zl\":8,\"zp\":{\"gs_ss\":\"1\"}}],[\"ペイペイカードゴールド\",0,[143,362],{\"zf\":33,\"zl\":8,\"zp\":{\"gs_ss\":\"1\"}}],[\"サッカー 浅野 弟\",0,[143,362],{\"zf\":33,\"zl\":8,\"zp\":{\"gs_ss\":\"1\"}}],[\"ポケモン sv かがやきパワーレシピ\",0,[143,362],{\"zf\":33,\"zl\":8,\"zp\":{\"gs_ss\":\"1\"}}],[\"ポケモン sv 夢特性入手方法\",0,[143,362],{\"zf\":33,\"zl\":8,\"zp\":{\"gs_ss\":\"1\"}}],[\"熊本地震速報震源地\",0,[143,362],{\"zf\":33,\"zl\":8,\"zp\":{\"gs_ss\":\"1\"}}],[\"スバル レヴォーグ\",0,[143,362],{\"zf\":33,\"zl\":8,\"zp\":{\"gs_ss\":\"1\"}}],[\"mlb news\",0,[143,362],{\"zf\":33,\"zl\":8,\"zp\":{\"gs_ss\":\"1\"}}],[\"帝京大学ゼミ男女差別教授\",0,[143,362],{\"zf\":33,\"zl\":8,\"zp\":{\"gs_ss\":\"1\"}}],[\"名鉄名古屋本線踏切事故\",0,[143,362],{\"zf\":33,\"zl\":8,\"zp\":{\"gs_ss\":\"1\"}}]],\"1\":{\"ag\":{\"a\":{\"8\":[\"話題の検索キーワード\"]}},\"k\":1,\"q\":\"b-8V15fiDAldvMgM8dUmVen0kpY\"}}"}
-----END CDP RESPONSE-----
-----START CDP ACTION-----
name: sessionStorage
args:
origin: "https://www.google.com"
-----END CDP ACTION-----
2022/11/25 13:33:04 sessionStorage: len: %d, item: %v 2 [_c;;i p:*|l:9007199254740991_2]
2022/11/25 13:33:04 sessionStorage: len: %d, item: %v 2 [hsb;;1669350782918 p:*|l:0_[1669350782917]]
2022/11/25 13:33:04 sessionStorage: len: %d, item: %v 2 [hsb;;1669350782917 p:*|l:1_{"state":null,"url":"/","metadata":{"mRa":1669350782916,"Dl":1669350782917,"Qja":1669350782918,"HG":0}}]
-----START CDP RESPONSE-----
name: sessionStorage
response:
items: {"_c;;i":"p:*|l:9007199254740991_2","hsb;;1669350782917":"p:*|l:1_{\"state\":null,\"url\":\"/\",\"metadata\":{\"mRa\":1669350782916,\"Dl\":1669350782917,\"Qja\":1669350782918,\"HG\":0}}","hsb;;1669350782918":"p:*|l:0_[1669350782917]"}
-----END CDP RESPONSE-----
<<<<<END CDP<<<<<
Browser Storage ... ok

1 scenario, 0 skipped, 0 failures
```

### CODE REVIEW

**Comment by atsushi-ishibashi** [comment]
I wonder how to test. 
Can I make runbook which access to `https://www.google.com/` ? I just check if you can get storage data.

**Comment by atsushi-ishibashi** [comment]
Btw should I use array(`[]struct{{"key": "<>", "value": "<>"}}`) instead of map.
because key value will be generated and I'd like to access map value with combined key.
But this doesn't work well.
```
desc: expr
vars:
  itemId: B
  items:
    A: a
    B: b
    C: c
steps:
  hoge:
    bind:
      itemName: vars.items["{{ vars.itemId }}"]
  fuga:
    dump: itemName
```

**Comment by k1LoW** [comment]
> But this doesn't work well.

How about the following?

```yml
desc: expr
vars:
  itemId: B
  items:
    A: a
    B: b
    C: c
steps:
  hoge:
    bind:
      itemName: vars.items[vars.itemId]
  fuga:
    dump: itemName
```

**Comment by k1LoW** [comment]
> https://github.com/k1LoW/runn/pull/307#issuecomment-1327017400

How about preparing HTML like the following and using scripts in it to store test data in localStorage or sessionStorage?

ref: https://github.com/k1LoW/runn/blob/94d00e8d7ee05e74aad4c1966ecd67b873738948/testutil/http.go

```html
<!doctype html>
<html>
<head>
  <title>For runn test</title>
</head>
<body>
  THIS IS BODY
<script>
  localStorage.setItem('key', 'value');
  sessionStorage.setItem('key', 'value');
</script>
</body>
</html>
```

**Comment by atsushi-ishibashi** [comment]
>How about preparing HTML like the following and using scripts in it to store test data in localStorage or sessionStorage?

good idea👍

And the runbook works well, thanks!

**Comment by atsushi-ishibashi** [comment]
I inserted script section so when you think I should avoid it, please let me know.

File: cdpfn.go [hunk 1]
```diff
@@ -237,6 +238,62 @@ var CDPFnMap = map[string]CDPFn{
...
+				}
+				*items = m
+				return nil
+			})
+		},
+		Args: CDPFnArgs{
+			{CDPArgTypeArg, "origin", "https://github.com"},
+			{CDPArgTypeRes, "items", `{"key": "value"}`},
+		},
+	},
```
Comment (k1LoW) [inline cdpfn.go:296]: Please add alias. 
You can update the documentation with `make doc`.

```suggestion
		Aliases: []string{"getSessionStorage"},
	},
```

**Review by k1LoW — APPROVED** [review]
Thank you!

**Pull request merged**